### PR TITLE
test_power_draw: wait for power instead of fixed warmup time

### PR DESCRIPTION
**Pull request by jnewb1**
- has to recieve 8 seconds of valid messages and valid power, with a timeout of 30 seconds

image

### CODE REVIEW

File: system/hardware/tici/tests/test_power_draw.py [hunk 1]
```diff
@@ -48,35 +50,78 @@ def setUp(self):
...
+    return int(sum(SAMPLE_TIME * SERVICE_LIST[msg].frequency for msg in proc.msgs))
+
+  def get_power_with_warmup_for_target(self, proc, prev):
+    socks = {msg: messaging.sub_sock(msg) for msg in proc.msgs}
+    for sock in socks.values():
+      messaging.drain_sock_raw(sock)
+
+    msgs_and_power = deque([None] * SAMPLE_TIME, maxlen=SAMPLE_TIME)
+
+    start_time = time.time()
```
Comment (adeebshihadeh) [inline system/hardware/tici/tests/test_power_draw.py:70]: `time.time` can jump, use `time.monotonic` instead

File: system/hardware/tici/tests/test_power_draw.py [hunk 2]
```diff
@@ -48,35 +50,75 @@ def setUp(self):
...
+
+    while (time.time() - start_time) < MAX_WARMUP_TIME:
+      power = get_power(1)
+      iteration_msg_counts = {}
+      for msg,sock in socks.items():
+        iteration_msg_counts[msg] = len(messaging.drain_sock_raw(sock))
+      msgs_and_power.append((power, iteration_msg_counts))
+
+      if any(m is None for m in msgs_and_power):
```
Comment (adeebshihadeh) [inline system/hardware/tici/tests/test_power_draw.py:79]: I think checking the length is more clear here

File: system/hardware/tici/tests/test_power_draw.py (hunk 1 above)
Comment (jnewb1) [reply system/hardware/tici/tests/test_power_draw.py]: done

File: system/hardware/tici/tests/test_power_draw.py (hunk 2 above)
Comment (jnewb1) [reply system/hardware/tici/tests/test_power_draw.py]: done

**Pull request merged**